web/src/widgetize.ts

```diff
@@ -42,23 +70,23 @@ export function activate(in_opts) {
         return;
     }
 
-    const callback = function (data) {
+    const callback = function (data: string): void {
         post_to_server({
             msg_type: "widget",
             data,
         });
     };
 
     if (
-        $row.attr("id").startsWith(`message-row-${message_lists.home.id}-`) &&
+        $row.attr("id")!.startsWith(`message-row-${message_lists.home!.id}-`) &&
         narrow_state.active()
     ) {
         // Don't place widget in a home message row if we are narrowed
         // to active state
         return;
     }
 
-    let $widget_elem = widget_contents.get(message.id);
+    let $widget_elem: JQuery | undefined = widget_contents.get(message.id);
```

Comment:
I think TS will automatically infer this given that `widget_contents` has been typed.

Reply:
yah...this seems redundant here..